src/poetry/plugins/plugin_manager.py

```diff
@@ -80,4 +100,219 @@ def _load_plugin_entry_point(self, ep: metadata.EntryPoint) -> None:
                 "The Poetry plugin must be an instance of Plugin or ApplicationPlugin"
             )
 
-        self.add_plugin(plugin())
+        self._add_plugin(plugin())
+
+
+class ProjectPluginCache:
+    PATH = Path(".poetry") / "plugins"
+
+    def __init__(self, poetry: Poetry, io: IO) -> None:
+        self._poetry = poetry
+        self._io = io
+        self._path = poetry.pyproject_path.parent / self.PATH
+        self._config_file = self._path / "config.toml"
+        self._gitignore_file = self._path / ".gitignore"
```

Comment:
I think this should point to `.poetry/.gitignore` so that if we use `.poetry` for another purpose in the future, we don't have to multiply the `.gitignore` files.

Reply:
done